lcUI/cadmdichild.cpp

```diff
@@ -217,7 +217,9 @@ void CadMdiChild::setDestroyCallback(LuaIntf::LuaRef destroyCallback) {
 
 void CadMdiChild::keyPressEvent(QKeyEvent *event) {
     QWidget::keyPressEvent(event);
-    emit keyPressed(event);
+    //Trap key here
+    if(event->key()==Qt::Key_Escape) emit escPressed();
```

Comment:
Will this be configurable?

Reply:
Actually, I am trying to change to UI to something like i said in zulip chat, i think capturing key presses here and clicommand line will help to obtain that feature.
It goes something like this.
When i press to C, it will go to add circle mode.
For UI, i am thinking of adding A left plane to show the possible operations next with first selected.
For Every selection, the type in plane will be updated as required and tab can be used for quick next. The button like p-represents points ,.. 
The mouse selection can be done for left plane(fallback);

I think this will increase the speed of working.